src/plugins/lua/replies.lua

```diff
@@ -34,42 +34,107 @@ local settings = {
   expire = 86400, -- 1 day by default
   key_prefix = 'rr',
   key_size = 20,
+  sender_prefix = 'rsrk',
+  sender_key_global = 'verified_senders',
+  sender_key_size = 20,
   message = 'Message is reply to one we originated',
   symbol = 'REPLY',
   score = -4, -- Default score
+  max_local_size = 20,
+  max_global_size = 30,
   use_auth = true,
   use_local = true,
   cookie = nil,
   cookie_key = nil,
   cookie_is_pattern = false,
   cookie_valid_time = '2w', -- 2 weeks by default
   min_message_id = 2, -- minimum length of the message-id header
+  reply_sender_privacy = false,
+  reply_sender_privacy_alg = 'blake2',
+  reply_sender_privacy_prefix = 'obf',
+  reply_sender_privacy_length = 16,
 }
 
 local N = "replies"
 
 local function make_key(goop, sz, prefix)
   local h = hash.create()
   h:update(goop)
-  local key
-  if sz then
-    key = h:base32():sub(1, sz)
-  else
-    key = h:base32()
-  end
+  local key = (prefix or '') .. h:base32():sub(1, sz)
+  return key
+end
 
-  if prefix then
-    key = prefix .. key
-  end
+local global_replies_set_script
+local local_replies_set_script
+
+local function configure_redis_scripts(_, _)
+  local redis_script_zadd_global = [[
+      -- getting last score of the recipients from the global replies set
+      local data = {}
+      data = redis.call('ZRANGE', KEYS[1], '-1', '-1', 'WITHSCORES')
+      local score = tonumber(data[#data])
+      if score == nil then
+        score = 0
+      end
 
-  return key
+      local global_size = redis.call('ZCARD', KEYS[1])
+      if global_size > {= mgs =} then
+        redis.call('ZREMRANGEBYRANK', KEYS[1], '{= mgs =}', global_size)
+      end
+
+      local recipients_addrs = ARGV
+      if recipients_addrs ~= nil then
+        local params = {}
+        --table.insert(params, KEYS[1])
+        for _, rcpt in ipairs(recipients_addrs) do
+          -- score is using here as a one-based numeration just like in lua
+          score = score + 1
+          table.insert(params, score)
+          table.insert(params, tostring(rcpt))
+        end
+        -- adding recipients to the global replies set
+        redis.call('ZADD', KEYS[1], unpack(params))
```

Comment:
I'm sorry, but I'm lost in this code. It looks incorrect to me I'm afraid. What we need to do is to call `ZINCRBY key recipient 1` for each recipient. We must also remove `ZRANGE` call, as it has no point here. If you want to have not LFU (least frequently used) logic but LRU (least recently used), then just use task timestamp as a value, hence, replacing all `ZINCRBY` with `ZADD key recipient <task_timestamp>`. However, LFU logic seems to be more preferrable here to my opinion (I'd also address this question to @fatalbanana).

Reply:
Then use another Redis Lua script for these purposes.

Reply:
Sorry, what do you mean by using another Redis Lua script? What is the problem with using it inside this redis script?